doc/api/child_process.markdown

```diff
@@ -729,10 +729,30 @@ an error but may have unforeseen consequences. Specifically, if the process
 identifier (PID) has been reassigned to another process, the signal will be
 delivered to that process instead which can have unexpected results.
 
+See `kill(2)` for general reference.
+
 Note that while the function is called `kill`, the signal delivered to the
-child process may not actually terminate the process.
+child process may not actually terminate the process. E.g. on `linux` child
+processes of child processes will not be terminated when attempting to kill
+their parent. This is likely to happen when running a new process in a shell
+or with use of the `shell` option of `ChildProcess`.
 
-See `kill(2)`
+```js
+'use strict';
+const spawn = require('child_process').spawn;
+
+let child = spawn('sh', ['-c',
+  `node -e "setInterval(() => {
+      console.log(process.pid + 'is alive')
+    }, 500);"`
+  ], {
+    stdio: ['inherit', 'inherit', 'inherit']
+  });
+
+setTimeout(() => {
+  child.kill(); // does not terminate the node process in the shell
```

Comment:
That would work. But for example not on OS X. The question then would be whether we want to document this.
I am making progress with the the `get_child_processes` thing I was mentioning. If libuv accepts it I would rather point to such a method then, suggesting a rationale: query all children, kill all children. This negative pid number thing on some unices seems very unexpressive to me.